pkg/tsdb/loki/loki.go

```diff
@@ -124,8 +125,14 @@ func callResource(ctx context.Context, req *backend.CallResourceRequest, sender
 		return err
 	}
 
+	contentType := rawLokiResponse.ContentType
+
+	if contentType == "" {
+		contentType = "application/json"
+	}
+
 	respHeaders := map[string][]string{
-		"content-type": {"application/json"},
+		"content-type": {contentType},
```

Comment:
I guess this is needed because the formatting API returns `text/plain` or something like that, but I don't really like this. This might be far fetched, but with this we are turning our datasource in a "raw" proxy, which would easily allow things like XSS to happen.

Reply:
Btw, Prometheus also returns JSON on their `/format_query` responses: https://prometheus.io/docs/prometheus/latest/querying/api/#formatting-query-expressions

Reply:
@svennergr this could be a topic to discuss in thursdays sync. try to find out why **text/plain** was decided on and if it could possibly be changed for consistency.